### ISSUE TITLE
Show all issues, not only the ones assigned to you

### ISSUE THREAD

**Issue by bfelbo** [Microsoft/vscode-github-issues-prs#41]
This looks amazing! However, for some repos I'm the only developer and I'd like to just see all the issues instead of having to first assign them all to me. This seems like a simple feature to implement?

**Comment by jhpratt** [comment]
@chrmarti Looks like it should be as simple as adding a checkbox in the settings to toggle [this line](https://github.com/Microsoft/vscode-github-issues-prs/blob/d66c205b3972726879338c8077e0ba0e689118a0/src/github-issues-prs.ts#L298). Would a PR be accepted?

**Comment by chrmarti** [comment]
@jhpratt Sounds good, PR appreciated!

**Comment by Stephanevg** [comment]
Any news on this guys? @chrmarti

**Comment by jhpratt** [comment]
@Stephanevg I'll get around to it at some point; it's by no means a priority of mine though. I've never touched a VS Code extension, so I honestly have no idea how to add in a checkbox.

**Comment by pmrt** [comment]
I was working on a PR for this but unfortunately it requires some changes:

https://github.com/microsoft/vscode-github-issues-prs/blob/8bca254b7ebd1e22a208f0d04554e38801b6808f/src/github-issues-prs.ts#L296-L310

Here we're fetching all the issues (visible for the provided Github account) and then filtering the results by repository. The problem is if we change **params.q** to something like this:

```javascript
let q = 'is:open';
const params = {
	q: this.showAll ? q : q += ` assignee:${assignee}`,
	order: 'asc',
	per_page: 100
};
```

When `params.q = 'is:open'` we're fetching all the issues visible for that account, which basically boils down to.. all issues in all the public repositories in the whole Github website. Yep, not good.

I suggest opening a new issue to change how it fetches all issues. Instead of fetching all first and then filtering the results, it should fetch only the ones for the repositories (ie. remotes) we need.

I found [this method](https://octokit.github.io/rest.js/#octokit-routes-issues-list-for-authenticated-user): **issues.listForAuthenticatedUsers()** in the library that this project is using. It lists all issues which the authenticated user owns or is member of. Maybe this method _**github.issues.listForAuthenticatedUser()**_ fits the project better than _**github.search.issues()**_ ?

**Comment by chrmarti** [comment]
`listForAuthenticatedUsers()` might be an optimization for the current query. Note that we later filter by repository (`remotes` parameter in this function).

To implement `showAll` you might need to run `fetchAll` for each repository. I originally did that and then realized that this runs into GitHub's rate limit for queries much earlier. It might still be the right thing to do here and it will work well when the list of `remotes` is not too long (it will usually be a single remote). I'd suggest adding an `if (showAll) { /* multiple queries */ } else { /* existing query */ }`.

**Comment by pmrt** [comment]
Exactly.

> To implement showAll you might need to run fetchAll for each repository

Yeah, that was my point. But just with the current implementation, not with `listForAuthenticatedUsers()`. The problem here is we're using `search.issue()` and thus fetching all visible issues _[for the user]_ in the whole github, filtering by assignee. When we don't filter by assignee we get all visible issues _[for the user]_ in the whole github (what turns out to be all issues of all public repositories). So yeah, by using `search.issue()` we would need to filter by the provided repositories (remotes), which would lead us to run `fetchAll` for each repository.

But by fetching with `listForAuthenticatedUsers()` we're already filtering by repositories the user owns/is member of (and then filtering the results again by the provided repositories in the `remotes` parameter), so we would fetch in one round trip all visible issues of just the needed repositories.

To sum up we have two options:
A) Opening a new issue to change the implementation of `fetchAllIssues()`using `issues.listForAuthenticatedUsers()` instead of `search.issues()`. Once it's finished, tackle #41 (this one)
B) Keeping the current implementation with `search.issues()` and just like @chrmarti suggested: `if (showAll) { /* multiple queries */ } else { /* existing query */ }`. This is the easiest and fastest way.

IMHO I would recommend to go for A.

**Comment by chrmarti** [comment]
I think `issues.listForAuthenticatedUsers()` with `filter: 'all'` would return too many issues for members of large projects (e.g., VS Code).

**Comment by pmrt** [comment]
Yeah, that's right. However, that's the point of this issue, to show all issues right? I mean, if you're member of large projects you wouldn't want to enable this feature anyway because you'd get lost with so many issues. 

What about limiting the results to X  pages to avoid performance issues? That way instead of checking with a ` if (showAll)` each iteration of` fetchAllIssues()` we would just fetch the results, limiting excessive amount of pages which could cause performance issues.  Is there any use case where the user needs the extension to display +200-300 issues?

If you can't find the issues you want you just provide the repositories you need in the configuration or just don't enable this option.
